xcode/upload-symbols-py3.py

```diff
@@ -185,19 +184,23 @@ def tar_gz_dsyms(search_path, single_file):
 
     search_root = os.path.abspath(search_path)
     if single_file:
-        dsyms = [os.path.basename(search_root)]
+        dsyms = [os.path.split(search_root)]
         search_root = os.path.dirname(search_root)
     else:
         log.debug("looking for dsyms in %s", search_root)
-        dsyms = [f for f in os.listdir(search_root) if f.endswith(".dSYM")]
+        dsyms = []
+        for root, dirs, files in os.walk(search_root):
+            dsyms.extend((root, d) for d in dirs if d.endswith(".dSYM"))
+            dsyms.extend((root, f) for f in files if f.endswith(".dSYM"))
```

Comment:
dsyms are bundles, I don't believe they should ever show up in the "files" list.